rfcs/followers.md

```diff
@@ -0,0 +1,106 @@
+RFC: Follower cluster peers
+===========================
+
```

Comment:
@hsanjuan This RFC is very implementation heavy and currently lack any context or definition of what we are attempting to implement. 
Could we get a draft of the following things:
- What a `follower` peer is?
- How a `follower` follows a cluster? Not the command invocation but logically.
- What part does a `follower` play in consensus, if any at all?
- A `follower` peer suggests there is a `non-follower` entity, what is it and what is its functionality/responsibilities?

This information will make it clear why, for example, we need different RPC auth policies etc. I think it is important that we get the non-implemenation details clarified so that we understand why the things in the rest of the doc are required and also so anyone outside of the dev team knows what this is about.

Sidenote: I would like to see the high-level specification of what follower peers are actually merged.

Reply:
So this is not user documentation. This builds on extensive previous discussions in the dev team and is for the dev team. Perhaps I've made a bad mistake using `followers` in the name because this is more about auth, shorthands and possibilities for initialization.

To recap: A follower is a `--consensus crdt` peer with defined `trusted peers`  and a strict rpc policy. It just like any other peer. A follower just an alias and a shorthand command to run a cluster peer with those things. I thought this was super clear at this point. There is nothing special in what a follower is vs. any other peer.

Now, we can align with the actual implemetation approaches here, we can maybe write proper explanations for everyone else and you can make the case for the nuances of not considering followers part of a "cluster" in the docs etc.

The main issue here is to settle on how to do `1.`.

Reply:
I am not advocating for user docs but for implementers docs, i.e. from an implementers perspective what is a follower peer.